packages-exp/auth-exp/src/platform_cordova/popup_redirect/popup_redirect.ts

```diff
@@ -20,26 +20,71 @@ import * as externs from '@firebase/auth-types-exp';
 import { browserSessionPersistence } from '../../platform_browser/persistence/session_storage';
 import { Auth } from '../../model/auth';
 import {
+  AuthEvent,
   AuthEventType,
-  EventManager,
   PopupRedirectResolver
 } from '../../model/popup_redirect';
 import { AuthPopup } from '../../platform_browser/util/popup';
-import { _fail } from '../../core/util/assert';
+import { _createError, _fail } from '../../core/util/assert';
 import { AuthErrorCode } from '../../core/errors';
 import {
   _checkCordovaConfiguration,
   _generateHandlerUrl,
   _performRedirect
 } from './utils';
-import { _generateNewEvent } from './events';
+import {
+  _eventFromPartialAndUrl,
+  _generateNewEvent,
+  _getAndRemoveEvent
+} from './events';
+import { AuthEventManager } from '../../core/auth/auth_event_manager';
+
+/**
+ * How long to wait for the initial auth event before concluding no
+ * redirect pending
```

Comment:
```suggestion
 * redirect pending.
```

Reply:
We don't enforce this anywhere and I don't particularly want to start :)